Modules/_functoolsmodule.c

```diff
@@ -748,25 +762,45 @@ typedef struct lru_list_elem {
     PyObject *key, *result;
 } lru_list_elem;
 
+static int
+lru_list_elem_clear(lru_list_elem *link)
```

Comment:
Looks like lru_list_elem have performance issue when supporting the GC in： https://github.com/python/cpython/pull/5008/files

Reply:
Heap types _must_ be fully GC compatible. See Victor's comments in the linked bpo. The alternative is to revert all heap type changes in functools, and leave it with static types.

Reply:
Hummm, in this case I think it makes sense to revert due to the memory and speed concerns, although someone should run some benchmarks.

It would be great to have also @methane's input here.

Reply:
Yes, we should have some updated benchmarks. The previous ones are from 2017.

Reply:
> Heap types _must_ be fully GC compatible. See Victor's comments in the linked bpo.

As my understanding, Victor said heap type object must fully GC compatible. Not its instance.
Am I wrong?

Reply:
I'm not sure I understand what you mean. If you're not visiting an instance of a type object, what are you visiting?

Reply:
This should be resolved by 45469a74f534136d9aabd748638e74a30af0f42d.

Reply:
But now it leaks refs. If I revert to the original PR, there's no ref. leaks.